File: app/src/main/java/com/lihb/library2d/adapter/BaseQuickWheelAdapter.java
package com.lihb.library2d.adapter;

import android.support.annotation.LayoutRes;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import java.util.ArrayList;
import java.util.List;

public abstract class BaseQuickWheelAdapter<T> extends AbstractWheelAdapter {
    private List<T> mData;
    private int itemResourceId;

    public BaseQuickWheelAdapter(@LayoutRes int itemLayoutId) {
        itemResourceId = itemLayoutId;
    }

    public BaseQuickWheelAdapter(@LayoutRes int itemLayoutId, List<T> data) {
        itemResourceId = itemLayoutId;
        mData = data;
    }

    public void setNewData(List<T> data) {
        mData = data;
    }

    public List<T> getData() {
        return mData == null ? new ArrayList<T>() : mData;
    }

    @Override
    public int getItemsCount() {
        return mData == null ? 0 : mData.size();
    }

    @Override
    public View getItem(int index, View convertView, ViewGroup parent) {
        if (index >= 0 && index < getItemsCount()) {
            if (convertView == null) {
                convertView = LayoutInflater.from(parent.getContext()).inflate(itemResourceId, parent, false);
            }
            onBindData(index, mData.get(index), convertView);
            return convertView;
        }
        return null;
    }

    protected abstract void onBindData(int index, T bean, View convertView);
}

File: wheelview/src/main/java/lihb/library/wheelview/BaseQuickWheelAdapter.kt
package lihb.library.wheelview

import android.support.annotation.LayoutRes
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import java.util.*

abstract class BaseQuickWheelAdapter<T> : AbstractWheelAdapter {
    private var mData: List<T>? = null
    private var itemResourceId: Int = 0

    val data: List<T>
        get() = if (mData == null) ArrayList() else mData!!

    override val itemsCount: Int
        get() = if (mData == null) 0 else mData!!.size

    constructor(@LayoutRes itemLayoutId: Int) {
        itemResourceId = itemLayoutId
    }

    constructor(@LayoutRes itemLayoutId: Int, data: List<T>) {
        itemResourceId = itemLayoutId
        mData = data
    }

    fun setNewData(data: List<T>) {
        mData = data
    }

    override fun getItem(index: Int, itemView: View?, parent: ViewGroup?): View? {
        var convertView = itemView
        if (index in 0..(itemsCount - 1)) {
            if (convertView == null) {
                convertView = LayoutInflater.from(parent!!.context).inflate(itemResourceId, parent, false)
            }
            onBindData(index, mData!![index], convertView!!)
            return convertView
        }
        return null
    }

    fun getItemBean(index: Int): T? {
        return if (mData != null && index >= 0 && index == itemsCount) {
            mData!![index]
        } else {
            null
        }
    }

    protected abstract fun onBindData(index: Int, bean: T, convertView: View)
}
